pkg/storage/stores/shipper/downloads/table.go

```diff
@@ -189,12 +189,24 @@ func (t *Table) DropUnusedIndex(ttl time.Duration, now time.Time) (bool, error)
 	var cleanedUpIndexSets []string
 
 	t.indexSetsMtx.RLock()
+	commonIndexSetExpired := false
 	for userID, userIndexSet := range t.indexSets {
 		lastUsedAt := userIndexSet.LastUsedAt()
 		if lastUsedAt.Add(ttl).Before(now) {
-			cleanedUpIndexSets = append(cleanedUpIndexSets, userID)
+			if userID == "" {
```

Comment:
I think worth adding a test @sandeepsukhani wdyt?. Just to make sure we are not missing anything here.